### ISSUE TITLE
Problem with macos build 

### ISSUE THREAD

**Issue by mchusovlianov** [chai2010/webp#21]
I try to build application which use webp library and can't build it with enabled CGO:
``` GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 go build -ldflags "-X main.Version=2.1.29 -X main.Build=`git rev-parse HEAD` -X main.BuildTime=`date +%FT%T%z`" -o bin ```
I have gcc installed.
![image](https://user-images.githubusercontent.com/196578/47353352-48fa8580-d6e6-11e8-9332-e81a6a74cc74.png)

**Comment by ns3777k** [comment]
facing the same issue. just curious. are you using go modules?

**Comment by solher** [comment]
Having the same issue here. I'm trying to migrate my project to go modules :/

**Comment by ns3777k** [comment]
if you take a look inside the vendor folder, you won't find `internal` directory where `webp.h` should be located.
since i commit my vendor directory, i just `git clone` this repo inside `vendor` directory.
there probably should be a better way =\

**Comment by solher** [comment]
I just found the solution actually. The go modules tool is not able to resolve C dependencies so it doesn't see that some `.c`/`.h` files are actually in the `internal` package of the `chai2010/webp` lib. Just copy it by hand in the vendor folder and it should work :)

**Comment by solher** [comment]
Hahaha perfect

**Comment by coldnight** [comment]
@ns3777k I'm facing same issue with `govendor`, with `govendor` you can install this package via 

```shell
﻿govendor fetch github.com/chai2010/webp/^
```

More details please refer to kardianos/govendor#137 and [here](https://github.com/kardianos/govendor#package-specifier).

**Comment by coldnight** [comment]
For now, Go module prunes non-package directories, see more discussions in golang/go#26366.

**Comment by coldnight** [comment]
Some useful tools:

- https://github.com/nomad-software/vend
- https://github.com/goware/modvendor

**Comment by kosta** [comment]
Is there a way to fix this bug on inside `chai2010/webp`? I feed like go vendoring is much less useful if you need to utilize external tools to make it work...

**Comment by coldnight** [comment]
I don't think we can find a way to fix this bug inside this project, this kind a issue(feature lacking) of `go mod`.

**Comment by pci** [comment]
For those coming across this issue who are using dep, a workaround is to add
```toml
[prune]
  go-tests = true
  unused-packages = true

  [[prune.project]]
    name = "github.com/chai2010/webp"
    go-tests = false
    unused-packages = false
```
to your Gopkg.toml to stop the pruning of the C code.

**Comment by furiousassault** [comment]
@chai2010, good day. Issue is still actual for go mod dependency management with vendoring.
Should we expect any changes that would make module to be downloaded with internals via go mod?
